### DISCUSSION TITLE
Issue with SPF alignment

### DISCUSSION THREAD

**Discussion by LANWrench** [postalserver/postal#1837] [Help with using Postal]
I am using a newly configured Postal server to test for replacing Microsoft SMTP in my small network.

We send mail on behalf of several different domains and it looked to me like Postal would be a good fit for what we are looking for.

The problem I am currently having is that I am unable to pass SPF alignment tests because the mailfrom domain doesn't match the from domain in the header.

Microsoft SMTP is working for us currently, here is an excerpt from the message header:
```
Authentication-Results: spf=pass (sender IP is x.x.x.x)
 smtp.mailfrom=example.com; dkim=none (message not signed)
 header.d=none;dmarc=bestguesspass action=none
 header.from=example.com;compauth=pass reason=109
```

However, from Postal, all the emails are going to spam because SPF isn't aligned. Here is an excerpt from the message header:
```
Authentication-Results: spf=pass (sender IP is x.x.x.x)
 smtp.mailfrom=postal.mail.myorg.com; dkim=pass (signature was verified)
 header.d=example.com;dmarc=bestguesspass action=none
 header.from=example.com;compauth=pass reason=109
```

Is there something in the configuration I am missing here?

Any help would be greatly appreciated.

**Comment by LANWrench** [comment]
Turns out one of my records was not right.

Been beating my head for hours on this and it wasn't until I posted this that I realized the problem.

Please disregard, thanks.

**Reply by LANWrench** [reply]
Well, it still shows this issue in spam assassin, but doesn't look to have a very heavy weight.

HEADER_FROM_DIFFERENT_DOMAINS | From and EnvelopeFrom 2nd level mail domains are different

**Reply by willpower232** [reply]
Are you making the message with SMTP? there may be an issue with how the message is being made and passed to Postal to send

Regarding your spf issue, whatsmydns.net is a good way of double checking your DNS records around the world.

**Reply by LANWrench** [reply]
Yes I am using SMTP to send (with powershell `send-mailmessage`).

I send the test mail through Microsoft SMTP and it delivers fine, with the headers above.
`Send-MailMessage -SmtpServer microsoftsmtpserver.myorg.com -Port 25 -From noreply@example.com -To myaddress@myorg.com -Subject test -Body test`

But get the other header above if sent through Postal:
`Send-MailMessage -SmtpServer postal.mail.myorg.com -Port 25 -From noreply@example.com -To myaddress@myorg.com -Subject test -Body test`

**Reply by willpower232** [reply]
Interesting, I'm afraid I don't know enough about powershells send-mailmessage to know what could be going wrong

**Reply by navinp** [reply]
> 

What's your Outbound spam threshold? Increase it.
